protocol/synthetix/contracts/storage/AccountDelegationIntents.sol

```diff
@@ -0,0 +1,149 @@
+ */
+library AccountDelegationIntents {
+    using SafeCastI256 for int256;
+    using SafeCastU128 for uint128;
+    using SafeCastU256 for uint256;
+    using SetUtil for SetUtil.UintSet;
+    using SetUtil for SetUtil.AddressSet;
+    using DelegationIntent for DelegationIntent.Data;
+
+    error InvalidAccountDelegationIntents();
+
+    struct Data {
+        uint128 accountId;
+        SetUtil.UintSet intentsId;
+        mapping(bytes32 => SetUtil.UintSet) intentsByPair; // poolId/collateralId => intentIds[]
+        // accounting for the intents collateral delegated
+        // Per Collateral
+        SetUtil.AddressSet delegatedCollaterals;
+        mapping(address => int256) netDelegatedAmountPerCollateral; // collateralType => net delegatedCollateralAmount
+    }
+
+    /**
+     * @dev Returns the account delegation intents stored at the specified account id.
+     */
+    function load(uint128 id) internal pure returns (Data storage accountDelegationIntents) {
+        bytes32 s = keccak256(abi.encode("io.synthetix.synthetix.AccountDelegationIntents", id));
+        assembly {
+            accountDelegationIntents.slot := s
+        }
+    }
+
+    /**
+     * @dev Returns the account delegation intents stored at the specified account id.
+     */
+    function loadValid(uint128 id) internal view returns (Data storage accountDelegationIntents) {
+        accountDelegationIntents = load(id);
+        if (accountDelegationIntents.accountId != 0 && accountDelegationIntents.accountId != id) {
+            revert InvalidAccountDelegationIntents();
+        }
+    }
+
+    /**
+     * @dev Returns the account delegation intents stored at the specified account id. Checks if it's valid
+     */
+    function getValid(uint128 id) internal returns (Data storage accountDelegationIntents) {
+        accountDelegationIntents = loadValid(id);
+        if (accountDelegationIntents.accountId == 0) {
+            // Uninitialized storage will have a 0 accountId
+            accountDelegationIntents.accountId = id;
+        }
+    }
+
+    function addIntent(Data storage self, DelegationIntent.Data storage delegationIntent) internal {
+        self.intentsId.add(delegationIntent.id);
+        self
+            .intentsByPair[
+                keccak256(abi.encodePacked(delegationIntent.poolId, delegationIntent.accountId))
+            ]
+            .add(delegationIntent.id);
+
+        self.netDelegatedAmountPerCollateral[delegationIntent.collateralType] += delegationIntent
+            .deltaCollateralAmountD18;
+
+        if (!self.delegatedCollaterals.contains(delegationIntent.collateralType)) {
+            self.delegatedCollaterals.add(delegationIntent.collateralType);
+        }
+    }
+
+    function removeIntent(
+        Data storage self,
+        DelegationIntent.Data storage delegationIntent
+    ) internal {
+        if (!self.intentsId.contains(delegationIntent.id)) {
+            return;
```

Comment:
Shouldn't we revert here? Since the guy is trying to remove an intent that doesn't exist?

Reply:
Didn't want to revert in this internal function since is used on another ones